### ISSUE TITLE
Inconsistent setParameter behavior for arrays

### ISSUE THREAD

**Issue by prideout** [google/filament#1270]
We declare uniform blocks with std140, which means that the element-to-element stride is always at least 16 bytes.

Currently, users are happy when they create arrays of `vec4` and `mat4`, and we provide template specializations for `mat3f` and `float3` in `UniformBuffer.cpp`.

However users are unhappy when they push values into arrays-of-float or arrays-of-float2.

We compute size and stride correctly, but we do not properly account for stride when users push values into material instances.

**tl;dr the `copy_n` in `UniformBuffer.cpp` isn't accounting for stride.**
